pkg/server/httpAPI.go

```diff
@@ -95,18 +96,24 @@ func (h *HTTPHandler) serveInitialize(rw http.ResponseWriter, req *http.Request)
 			h.initializationStatus = initializationStatusSuccessful
 		}()
 	}
-	rw.WriteHeader(http.StatusOK)
+	rw.WriteHeader(http.StatusAccepted)
 }
 
 // ServeInitializationStatus serves the etcd initialization progress status
 func (h *HTTPHandler) serveInitializationStatus(rw http.ResponseWriter, req *http.Request) {
-	//fmt.Fprintf(rw, "initialization request received.")
 	h.initializationStatusMutex.Lock()
 	defer h.initializationStatusMutex.Unlock()
 	h.Logger.Infof("Responding to status request with: %s", h.initializationStatus)
 
-	rw.WriteHeader(http.StatusOK)
-	rw.Write([]byte(h.initializationStatus))
+	rw.WriteHeader(http.StatusAccepted)
+	// If the handler status is 202 then the Snapshotter is being initialized.
+	// Return in progress till the snapshotter is being intialized and then
+	// on successful initialization set the initializationStatus to Success.
+	if h.Status == http.StatusAccepted {
```

Comment:
Not required. Please do not bring any dependency on `handler.Status` in initialisation API. Initialisation API logic is all good.